diff -Naur ffmpeg-3.3.7-old/libswresample/arm/audio_convert_neon.S ffmpeg-3.3.7-new/libswresample/arm/audio_convert_neon.S
--- ffmpeg-3.3.7-old/libswresample/arm/audio_convert_neon.S	2016-03-29 02:25:33.000000000 +0000
+++ ffmpeg-3.3.7-new/libswresample/arm/audio_convert_neon.S	2018-05-08 18:26:20.907508944 +0000
@@ -134,9 +134,9 @@
         itt             lt
         ldrlt           r1,  [r1]
         blt             X(swri_oldapi_conv_flt_to_s16_neon)
-        beq             X(swri_oldapi_conv_fltp_to_s16_2ch_neon)
-
-        push            {r4-r8, lr}
+	bne		1f
+        b               X(swri_oldapi_conv_fltp_to_s16_2ch_neon)
+1:	push            {r4-r8, lr}
         cmp             r3,  #4
         lsl             r12, r3,  #1
         blt             4f
