diff -Naur gst-plugins-ugly-old/configure.ac gst-plugins-ugly-new/configure.ac
--- gst-plugins-ugly-old/configure.ac	2017-12-07 15:47:42.000000000 +0000
+++ gst-plugins-ugly-new/configure.ac	2018-04-30 18:03:23.104733393 +0000
@@ -10,7 +10,7 @@
 AG_GST_INIT
 
 dnl initialize automake
-AM_INIT_AUTOMAKE([-Wno-portability 1.14 no-dist-gzip dist-xz tar-ustar subdir-objects])
+AM_INIT_AUTOMAKE([-Wno-portability 1.13 no-dist-gzip dist-xz tar-ustar subdir-objects])
 
 dnl define PACKAGE_VERSION_* variables
 AS_VERSION
